panel-plugin/i3w-config.c

```diff
@@ -75,22 +75,23 @@ serialize_gdkcolor(GdkColor *gdkcolor)
     return color;
 }
 
-GdkColor *
-unserialize_gdkcolor(guint32 color)
+GdkRGBA *
+unserialize_gdkrgba(guint32 color)
 {
-    GdkColor *gdkcolor = g_new0(GdkColor, 1);
+    GdkRGBA *gdkrgba = g_new0(GdkRGBA, 1);
 
     // Mask and shift the color components
-    guint16 red_component = (guint16) ((color & 0xff0000) >> 16);
-    guint16 green_component = (guint16) ((color & 0x00ff00) >> 8);
-    guint16 blue_component = (guint16) ((color & 0x0000ff));
+    guint8 red_component = (color & 0xff0000) >> 16;
+    guint8 green_component = (color & 0x00ff00) >> 8;
+    guint8 blue_component = (color & 0x0000ff);
 
-    // expand the GdkColor components to 16 bit
-    gdkcolor->red = red_component << 8;
-    gdkcolor->green = green_component << 8;
-    gdkcolor->blue = blue_component << 8;
+    // convert back to floats in range 0.0 to 1.0
+    gdkrgba->red = red_component / 255.0;
+    gdkrgba->green = green_component / 255.0;
+    gdkrgba->blue = blue_component / 255.0;
```

Comment:
`/ 255.0` is not equivalent to `<< 8`. You need to use `256` to get the same result, but what is wrong with using shifts?